synapse/tests/test_axon.py

```diff
@@ -1167,3 +1168,46 @@ async def test_axon_mirror(self):
 
             (size, sha256) = await axon01.put(b'vertex')
             self.eq(await axon00.size(sha256), await axon01.size(sha256))
+
+    async def test_axon_read_unpack(self):
+
+        async with self.getTestAxon() as axon:
+
+            user = await axon.auth.addUser('user')
+            await user.setPasswd('test')
+            _, port = await axon.dmon.listen('tcp://127.0.0.1:0')
+            aurl = f'tcp://user:test@127.0.0.1:{port}/axon'
+            async with await s_telepath.openurl(aurl) as prox:  # type: s_axon.AxonApi
+                await self.asyncraises(s_exc.AuthDeny, prox.read(asdfhash, 3))
+                await self.asyncraises(s_exc.AuthDeny, prox.unpack(asdfhash, '>Q'))
+                await user.addRule((True, ('axon', 'upload',)))
+                await user.addRule((True, ('axon', 'get',)))
+
+                data = b'vertex.link'
+                size, sha256 = await prox.put(data)
+                self.eq(b'tex', await prox.read(sha256, 3, offset=3))
+                self.eq(b'link', await prox.read(sha256, 4, offset=7))
+                self.eq(b'', await prox.read(sha256, 1, offset=11))
+
+                with self.raises(s_exc.BadArg):
+                    await prox.read(sha256, 0)
+                with self.raises(s_exc.BadArg):
+                    await prox.read(sha256, 1, -1)
+                with self.raises(s_exc.BadArg):
+                    await prox.read(sha256, 2 * 1024 * 1024)
+
+                intdata = struct.pack('>QQQ', 1, 2, 3)
```

Comment:
Can we add the struct tests to `runAxonTestBase` ? That way we'll ensure we're testing against the default implementation, telepath api, and downstream implementations.